api/dbadapters/bigquery.ts

```diff
@@ -312,14 +312,18 @@ export class BigQueryDbAdapter implements IDbAdapter {
   private getClient(projectId?: string) {
     projectId = projectId || this.bigQueryCredentials.projectId;
     if (!this.clients.has(projectId)) {
+      let bigQueryOptions = {
+        projectId,
+        scopes: EXTRA_GOOGLE_SCOPES,
+        location: this.bigQueryCredentials.location,
+        credentials: <any>this.bigQueryCredentials.credentials,
```

Comment:
i don't really understand this part - the field is optional, why set it in this case at all?

perhaps the compiler is complaining that you're setting a field that doesn't exist, in which case, i propose either of the following:

preferred:
```
new BigQuery({
  projectId,
  credentials: this.bigQueryCredentials.credentials && JSON.parse(this.bigQueryCredentials.credentials),
  scopes: EXTRA_GOOGLE_SCOPES,
  location: this.bigQueryCredentials.location
})
```

or: 
```
const bigqueryOptions: BigQueryOptions = {
  projectId,
  scopes: EXTRA_GOOGLE_SCOPES,
  location: this.bigQueryCredentials.location,
}
// etc
```